src/components/NewDatePicker/index.js

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import {View, Animated} from 'react-native';
+import moment from 'moment';
+import _ from 'underscore';
+import TextInput from '../TextInput';
+import CalendarPicker from '../CalendarPicker';
+import CONST from '../../CONST';
+import styles from '../../styles/styles';
+import * as Expensicons from '../Icon/Expensicons';
+import {propTypes as datePickerPropTypes, defaultProps as defaultDatePickerProps} from './datePickerPropTypes';
+
+const propTypes = {
+    ...datePickerPropTypes,
+};
+
+const datePickerDefaultProps = {
+    ...defaultDatePickerProps,
+};
+
+class NewDatePicker extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isPickerVisible: false,
+            selectedDate: new Date(props.value || props.defaultValue || undefined),
```

Comment:
```suggestion
            selectedDate: new Date(props.value || props.defaultValue || null),
```
I found that `undefined` and `null` have different results on Date but same results when using moment. Let's use `null` as that's what works for both.

Reply:
using moment of Date is not a problem. Just make sure the fallback parameter is `null` this will cover future cases of blind copy-paste.